### DISCUSSION TITLE
This module is declared with using 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.

### DISCUSSION THREAD

**Discussion by kelseyp99** [stx-labs/stacks.js#1186] [Q&A]
I am getting the following error with the Stacks.js when compiling tsc:
This module is declared with using 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.
Can anyone offer any solutions?  I am stuck.  This is my post below
"https://stackoverflow.com/questions/71082270/error-ts1259-module-node-modules-types-bn-js-index-can-only-be-default

**Comment by janniks** [comment]
Hi. Can you provide more details on your setup? Is this a TS node project or a frontend project (e.g. with React)?

**Comment by kelseyp99** [comment]
[image: image.png]

it is a node project.  I took the code directly from the github

On Tue, Feb 15, 2022 at 4:20 AM janniks ***@***.***> wrote:

> Hi. Can you provide more details on your setup? Is this a TS node project
> or a frontend project (e.g. with React)?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hirosystems/stacks.js/discussions/1186#discussioncomment-2179477>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABQPTAADC5CIHCNTILN6HHDU3ILGHANCNFSM5ONHJA7A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Phil

**Reply by kelseyp99** [reply]
Appears to have solved the problem.   See the stackoverflow above.  Thanks

**Reply by janniks** [reply]
Which node version are you using? We would love to fix this problem. 😄

**Reply by kelseyp99** [reply]
This is what I ran last. 
npm update @stacks/transactions@^2.0.1
npm install @stacks/network
How do I check the version?

**Reply by janniks** [reply]
Running `node -v` will output the used node version into the terminal.
Also try using the latest version of the `stacks.js` packages. e.g. `@stacks/transactions@^3.2.0`

**Comment by kelseyp99** [comment]
node version: v16.13.1

On Wed, Feb 16, 2022 at 8:06 AM janniks ***@***.***> wrote:

> Running node -v will output the used node version into the terminal.
> Also try using the latest version of the stacks.js packages. e.g.
> @stacks/transactions@^3.2.0
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hirosystems/stacks.js/discussions/1186#discussioncomment-2188437>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABQPTAAYAQ4BO26NSLBUAELU3OOONANCNFSM5ONHJA7A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Phil

**Reply by janniks** [reply]
Could you try again after updating with `npm install @stacks/transactions@^3.2.0` ?

**Comment by kelseyp99** [comment]
I updated by running
npm update @stacks/transactions@^3.2.0

I get the below two errors now.  The second I was getting before also.  Can
you tell me how to fix this?

from my tsconfig =>  "target": "es2020"

> tsc Scripts\shell.ts
Scripts/shell.ts:18:29 - error TS2737: BigInt literals are not available
when targeting lower
than ES2020.,

18 const postConditionAmount = 1000000n;
                               ~~~~~~~~

Scripts/shell.ts:35:21 - error TS1378: Top-level 'await' expressions are
only allowed when the 'module' option is set to 'es2022', 'esnext',
'system', or 'nodenext', and the 'target' option is set to 'es2017' or
higher.

35 const transaction = await makeContractCall(txOptions);
                       ~~~~~

Scripts/shell.ts:40:27 - error TS1378: Top-level 'await' expressions are
only allowed when the 'module' option is set to 'es2022', 'esnext',
'system', or 'nodenext', and the 'target' option is set to 'es2017' or
higher.

40 const broadcastResponse = await broadcastTransaction(transaction,
network);
                             ~~~~~

On Wed, Feb 16, 2022 at 10:21 AM Phil Kelsey ***@***.***> wrote:

> node version: v16.13.1
>
> On Wed, Feb 16, 2022 at 8:06 AM janniks ***@***.***> wrote:
>
>> Running node -v will output the used node version into the terminal.
>> Also try using the latest version of the stacks.js packages. e.g.
>> @stacks/transactions@^3.2.0
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/hirosystems/stacks.js/discussions/1186#discussioncomment-2188437>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ABQPTAAYAQ4BO26NSLBUAELU3OOONANCNFSM5ONHJA7A>
>> .
>> Triage notifications on the go with GitHub Mobile for iOS
>> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
>> or Android
>> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>>
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>
>
> --
> Phil
>


-- 
Phil
